engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1495,6 +1509,11 @@ private void validateThreadLocal(OperatorMeta om) {
       return;
     }
 
+    if (om.getOperator() instanceof Operator.DelayOperator) {
+      String msg = String.format("Locality %s invalid for delay operator %s", Locality.THREAD_LOCAL, om);
+      throw new ValidationException(msg);
```

Comment:
Yes, but since the DelayOperator can actually go back to itself, this will have some complication.  We will address this in the future.